webroot/src/components/Forms/InputPassword/InputPassword.ts

```diff
@@ -0,0 +1,91 @@
+//
+//  InputPassword.ts
+//  InspiringApps modules
+//
+//  Created by InspiringApps on 4/22/2020.
+//  Copyright © 2024. InspiringApps. All rights reserved.
+//
+
+import {
+    Component,
+    mixins,
+    Prop,
+    toNative
+} from 'vue-facing-decorator';
+import MixinInput from '@components/Forms/_mixins/input.mixin';
+import ShowPasswordEye from '@components/Icons/ShowPasswordEye/ShowPasswordEye.vue';
+import HidePasswordEye from '@components/Icons/HidePasswordEye/HidePasswordEye.vue';
+
+@Component({
+    name: 'InputPassword',
+    components: {
+        ShowPasswordEye,
+        HidePasswordEye
+    }
+})
+class InputPassword extends mixins(MixinInput) {
+    @Prop({ required: true }) joiMessages!: any;
+    @Prop({ default: false }) showEyeIcon?: boolean;
+
+    //
+    // Data
+    //
+    inputType = 'password';
+    shouldHidePassword = true;
+
+    //
+    // Computed
+    //
+    get passwordRequirements(): Array<any> {
+        // @NOTE: This is dipping into Joi private API - here be dragons. If this becomes a hassle, we can just manually add the "*" to the labels for now.
```

Comment:
Yeah, this is still needed. It's generic logic to grab the password-requirement messages from the input validations.

It would be used (e.g.) on a custom login UI or if we allow authenticated users to update their password.